src/OrbitGl/CaptureViewElementTest.cpp

```diff
@@ -67,4 +100,85 @@ TEST(CaptureViewElementTesterTest, PassesAllTestsOnExistingElement) {
                                                      tester.GetLayout(), kChildCount);
   tester.RunTests(&container_elem);
 }
+
+const Viewport kViewport(100, 100);
+const TimeGraphLayout kLayout;
+
+TEST(CaptureViewElement, IsMouseOverWorksForVisibleElements) {
+  UnitTestCaptureViewLeafElement elem(nullptr, &kViewport, &kLayout);
+  elem.SetWidth(kViewport.GetWorldWidth());
+
+  EXPECT_FALSE(elem.IsMouseOver(Vec2(-1, -1)));
+  EXPECT_TRUE(elem.IsMouseOver(Vec2(5, 5)));
+  EXPECT_TRUE(elem.IsMouseOver(elem.GetSize()));
+  EXPECT_TRUE(elem.IsMouseOver(Vec2(0, 0)));
+}
+
+TEST(CaptureViewElement, IsMouseOverDoesNotWorkForElementsOutsideOfTheParent) {
+  const int kChildCount = 2;
+  UnitTestCaptureViewContainerElement elem(nullptr, &kViewport, &kLayout, kChildCount);
+
+  CaptureViewElement* child0 = elem.GetAllChildren()[0];
+
+  static_assert(kLeafElementHeight > 5);
+  child0->SetPos(0, -5);
+  EXPECT_TRUE(child0->IsMouseOver(Vec2(0, 0)));
+  EXPECT_FALSE(child0->IsMouseOver(Vec2(0, -1)));
+}
+
+TEST(CaptureViewElement, MouseWheelEventRecursesToCorrectChildren) {
+  using ::testing::_;
+  using ::testing::Exactly;
+  using ::testing::Return;
+
+  const int kChildCount = 2;
+  UnitTestCaptureViewContainerElement container_elem(nullptr, &kViewport, &kLayout, kChildCount);
+
+  static_assert(kMarginAfterChild > 0);
+
+  const Vec2 kPosOutside(-1, -1);
+  const Vec2 kPosOnParent(10, kLeafElementHeight + 1);
+  const Vec2 kPosOnChild0(10, kLeafElementHeight);
+  const Vec2 kPosOnChild1(10, kLeafElementHeight + kMarginAfterChild);
+  const Vec2 kPosOnChild2(10, (kLeafElementHeight + kMarginAfterChild) * 2);
+
+  class ElementThatHandlesMouseWheelEvents : public UnitTestCaptureViewLeafElement {
+   public:
+    explicit ElementThatHandlesMouseWheelEvents(CaptureViewElement* parent,
+                                                const Viewport* viewport,
+                                                const TimeGraphLayout* layout)
+        : UnitTestCaptureViewLeafElement(parent, viewport, layout) {}
+
+   protected:
+    [[nodiscard]] bool OnMouseWheel(const Vec2& mouse_pos, int delta,
+                                    const ModifierKeys& modifiers) override {
+      return true;
```

Comment:
Why don't we mock that?

Reply:
I want this class to explicitly return `true` because it should fake a child that handles mouse wheel events - or am I missing your point?

Reply:
I was more concerned about the fact that there is a class definition showing up inside a function. I would rather create a Mock of that class above. And then inside this TEST function, state that your mock should return `true`.

Reply:
As discussed offline - I've removed the class here because the parent mock class can already handle this. Thanks for the help!